Sources/MongoSwift/ReadConcern.swift

```diff
@@ -14,65 +14,94 @@ public class ReadConcern: Codable {
+                self = .available
+            case "majority":
+                self = .majority
+            case "linearizable":
+                self = .linearizable
+            default:
+                self = .custom(level: rawValue)
+            }
         }
-        return String(cString: level)
     }
 
+    /// The level of this `ReadConcern`, or `nil` if the level is not set.
+    public var level: Level?
+
     /// Indicates whether this `ReadConcern` is the server default.
     public var isDefault: Bool {
-        return mongoc_read_concern_is_default(self._readConcern)
+       if level == nil {
+           return true
+       }
+       return false
     }
 
-    /// Initialize a new `ReadConcern` from a `ReadConcern.Level`.
-    public convenience init(_ level: Level) {
-        self.init(level.rawValue)
+    /// Initializes a new `ReadConcern` by copying a `mongoc_read_concern_t`.
+    /// The caller is responsible for freeing the original `mongoc_read_concern_t`.
+    public init(from readConcern: OpaquePointer) {
+        if let level = mongoc_read_concern_get_level(readConcern) {
+            self.level = Level(rawValue: String(cString: level))
+        }
+    }
+
+    /// Initializes a new `ReadConcern` by copying an existing `ReadConcern`.
+    public init(from readConcern: ReadConcern) {
+        self.level = readConcern.level
+    }
+
+    /// Initialize a new `ReadConcern` with a `Level`.
+    public init(_ level: Level) {
+        self.level = level
     }
 
     /// Initialize a new `ReadConcern` from a `String` corresponding to a read concern level.
     public init(_ level: String) {
-        self._readConcern = mongoc_read_concern_new()
-        mongoc_read_concern_set_level(self._readConcern, level)
+        self.level = Level(rawValue: level)
     }
 
     /// Initialize a new empty `ReadConcern`.
     public init() {
-        self._readConcern = mongoc_read_concern_new()
+        self.level = nil
     }
 
     /// Initializes a new `ReadConcern` from a `Document`.
-    public convenience init(_ doc: Document) {
+    public init(_ doc: Document) {
         if let level = doc["level"] as? String {
             self.init(level)
         } else {
             self.init()
         }
     }
 
-    /// Initializes a new `ReadConcern` by copying an existing `ReadConcern`.
-    public init(from readConcern: ReadConcern) {
-        self._readConcern = mongoc_read_concern_copy(readConcern._readConcern)
-    }
-
-    /// Initializes a new `ReadConcern` by copying a `mongoc_read_concern_t`.
-    /// The caller is responsible for freeing the original `mongoc_read_concern_t`.
-    internal init(from readConcern: OpaquePointer?) {
-        self._readConcern = mongoc_read_concern_copy(readConcern)
-    }
-
     private enum CodingKeys: String, CodingKey {
         case level
     }
 
-    public required convenience init(from decoder: Decoder) throws {
+    public init(from decoder: Decoder) throws {
```

Comment:
Since `level` is now a stored property rather than computed, is it possible to delete this `init` and take advantage of the compiler-synthesized implementation?

Reply:
same question applies about the `encode` method.